### PR TITLE
Search result pagination

### PR DESCRIPTION
**Pull request by stefanodigirolamo**
#### What problem is this solving?

The implementation will solve some SEO problems related to the exsisting Fetch More pagination flow 

#### How to test it?

It's possible to test it in every product list page 

https://stefanosprint14--ruwhirlpool.myvtex.com/

#### Screenshots or example usage:

BEFORE:

image

AFTER:

image

### CODE REVIEW

**Comment by kaibrockelt** [comment]
Hi! 
Why set me as a reviewer? im not a developer, and most likely have low enough code understanding to make this project a living hell. Better give this into the hands of a dev person who knows what they are doing on that project.

**Comment by stefanodigirolamo** [comment]
> Hi! Why set me as a reviewer? im not a developer, and most likely have low enough code understanding to make this project a living hell. Better give this into the hands of a dev person who knows what they are doing on that project.

Sorry i don't why it automatically add you as reviewer.

**Comment by kaibrockelt** [comment]
No worries!
Just making space for the Pro’s :)

Kai Brockelt
www.vtex.com
Customer Experience

Passeig de Gracia, 13
Planta 3, 08007
Barcelona
____________________________________________

**Comment by hiagolcm** [comment]
@stefanodigirolamo, thanks for the contribution 😄 

Could you explain the SEO problems you are facing?

We don't have this pagination style because it's not recommended by the [Baymard Institute](https://baymard.com/)

Here are some quotes from it

https://baymard.com/premium/guideline-collections/uctfzk/501
> “It’s not really moving on.” Pagination links on mobile devices are often difficult to tap successfully due to their size and proximity to other links.
Throughout multiple rounds of testing, numerous subjects complained about pagination. They generally perceived pagination to be slow, and seeing more than a handful of pagination links would often discourage them from browsing the full product list

https://baymard.com/premium/guideline-collections/cb3zzh/1777
> Our large-scale, general usability testing of these three loading schemas revealed that endless scrolling performs the worst for users, pagination performs slightly better, and “Load More” performs the best it comes to general usability

If what you are looking for is just the "progress feeling", you can use the [search-products-progress-bar](https://github.com/vtex-apps/search-result/tree/master/docs#the-search-products-progress-bar-block). It will add a progress bar to the pagination button.

image

**Comment by stefanodigirolamo** [comment]
> 
@hiagolcm , thanks for your answer.

I created this PR in order to fix a specific SEO problem and also for implement a numeric pagination asked by the business team of the project where i work. 
The SEO problem we faced is related to the pagination indexing, in fact if you i'll try to check the DOM there are no direct links to all the pages of a product list.
The progress-bar suggestion is a fine decision to show user how many products we have, but unfortunately it couldn't help search robots to index correctly next pages of listing.
I remain at your disposal for any doubts or clarification.

File: react/hooks/useFetchMore.js [hunk 1]
```diff
@@ -185,6 +197,141 @@ export const useFetchMore = (props) => {
     isFirstRender.current = false
   }, [maxItemsPerPage, query, map, orderBy, priceRange])
 
+  //PAGINATION
```
Comment (hiagolcm) [inline react/hooks/useFetchMore.js:200]: I don't think the way you implemented the fetch will help you with the SEO problem.

The buttons you created will have the same behavior as the "See More" button. You are just fetching it using javascript. I think Google recommends to use the history API

I would recommend checking the [render-runtime](https://github.com/vtex-apps/render-runtime) project and looking for the `Link` component and the `navigate` function. Check if they use the history API

**Review by hiagolcm — CHANGES_REQUESTED** [review]

**Pull request closed without merging**